File: hospital.client/src/types/PrescriptionResponse.ts
import type { PrescriptionItemResponse } from "./PrescriptionItemResponse";

export interface PrescriptionResponse {
  id: number;
  consultationId: number;
  doctorId: number;
  prescriptionDate: string;
  notes?: string | null;
  state: number;
  createdAt: string;
  createdBy: number;
  updatedBy?: number | null;
  updatedAt?: string | null;
  items?: PrescriptionItemResponse[] | null;
}

export interface PrescriptionRequest {
  id?: number | null;
  consultationId?: number | null;
  doctorId?: number | null;
  prescriptionDate?: string | null;
  notes?: string | null;
  state?: number | null;
  createdBy?: number | null;
  updatedBy?: number | null;
  /** Items to create together with the prescription (used in createPrescriptionWithItems) */
  items?: PrescriptionItemInlineRequest[] | null;
}

export interface PrescriptionItemInlineRequest {
  medicineName: string;
  dosage: string;
  frequency: string;
  duration: string;
  specialInstructions?: string | null;
}

File: Hospital.Server/Entities/Response/PrescriptionResponse.cs
namespace Hospital.Server.Entities.Response
{
    public class PrescriptionResponse
    {
        public long Id { get; set; }
        public long ConsultationId { get; set; }
        public long DoctorId { get; set; }
        public string PrescriptionDate { get; set; } = string.Empty;
        public string? Notes { get; set; }
        public int State { get; set; }
        public string CreatedAt { get; set; } = string.Empty;
        public long CreatedBy { get; set; }
        public long? UpdatedBy { get; set; }
        public string? UpdatedAt { get; set; }

        /// <summary>Items included when fetched with include=Items</summary>
        public virtual ICollection<PrescriptionItemResponse>? Items { get; set; }
    }
}
